### DISCUSSION TITLE
Refactoring Brownie Testing

### DISCUSSION THREAD

**Discussion by JohnEMagee** [smartcontractkit/full-blockchain-solidity-course-py#1212] [General]
I come from the ruby on rails / capybara / rspec world, and we don't really like repeating ourselves if we don't have to.

Can repeated action (i.e. deploying the test contract) be extracted out of all tests and set up to execute before all tests?

**Comment by cromewar** [comment]
Hello @JohnEMagee 
Yes it can, you can create a python folder an set up the functions you know you are going to use over and over again, and then just import it into your test script.

**Comment by JohnEMagee** [comment]
This isn’t exactly what I meant. 

In rspec/capybara you can set up a ‘before’ action at the top of a file for each test to run…in this case, the contract deployment would only have to be written once and called before each test is run automatically?

My better more complete experience is ruby/rails - python is a bit newer (and more awkward to be honest) to me

**Reply by cromewar** [reply]
May be this could be handy: https://levelup.gitconnected.com/advanced-pytest-techniques-i-learned-while-contributing-to-pandas-7ba1465b65eb
